aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -100,12 +108,9 @@ void CUDAGraph::capture_begin(MempoolId_t pool/*=0*/, cudaStreamCaptureMode capt
   auto* gen = get_generator_or_default<CUDAGeneratorImpl>(
       c10::nullopt, cuda::detail::getDefaultCUDAGenerator());
 
-  auto options = TensorOptions().device(at::kCUDA).dtype(at::kLong);
-  seed_extragraph_ = at::empty({1}, options);
-  offset_extragraph_ = at::empty({1}, options);
-
-  seed_extragraph_.fill_(int64_t(gen->current_seed()));
-  gen->capture_prologue(seed_extragraph_.data_ptr<int64_t>(), offset_extragraph_.mutable_data_ptr<int64_t>());
+  auto gen_ptr =
+      c10::intrusive_ptr<CUDAGeneratorImpl>::unsafe_reclaim_from_nonowning(gen);
```

Comment:
I'm unsure of the correct approach to reference the default `CUDAGeneratorImpl`. Currently, `get_generator_or_default` returns a raw pointer, but for safety, it seems better to have the graph hold a shared pointer instead. Therefore, I've used `unsafe_reclaim_from_nonowning` to convert this, though I'm open to a safer, alternative method.

Reply:
This seems acceptable, and what this API was intended for